### PR TITLE
Add support for 3DS v2.x

### PR DESCRIPTION
**Pull request by daneastmond**

### CODE REVIEW

File: src/Message/PurchaseRequest.php [hunk 1]
```diff
@@ -1495,14 +1820,35 @@ public function getData()
...
                 'threeDSReqAuthTimestamp' => $this->get3DsRequestAuthenticationTime(),
             ]);
+            // for optional DS_MERCHANT_EMV3DS param object
             $emv3DsParameters = array_filter([
+                // required parameters for v2
+                'browserAcceptHeader'                => $this->getBrowserAcceptHeader(),
+                'browserColorDepth'                  => $this->getBrowserColorDepth(),
+                'browserIP'                          => $this->getBrowserIP(),
+                'browserJavaEnabled'                 => $this->getBrowserJavaEnabled(),
+            //    'browserJavascriptEnabled'           => '', // conditional
```
Comment (samami) [inline src/Message/PurchaseRequest.php]: Looks like could be removed

File: src/Message/PurchaseRequest.php [hunk 2]
```diff
@@ -1495,14 +1820,35 @@ public function getData()
...
+                'protocolVersion'                    => $this->getProtocolVersion(),
+                // Indicates whether the 3DSMethod has been executed. Values ​​accepted: Y= Successfully completed, N = Completed with errors, U = 3DSMethod not executed
+                'threeDSCompInd'                     => 'U',
+                // Type of request. Possible values: CardData, AuthenticationData, ChallengeResponse
+                'threeDSInfo'                        => 'CardData',
+                'threeDSServerTransID'               => $this->getTransactionId(),
+                // optional parameters for v2
                 'cardholderName'                     => $this->getCardholder(),
-                'email'                              => $card->getEmail(),
+                'email'                              => $card->getEmail(), // in docs as 'E-mail'
```
Comment (samami) [inline src/Message/PurchaseRequest.php]: Which docs, and this should be on the previous line

**Review by samami — APPROVED** [review]

File: src/Message/PurchaseRequest.php (hunk 1 above)
Comment (daneastmond) [reply src/Message/PurchaseRequest.php]: Yep

File: src/Message/PurchaseRequest.php (hunk 2 above)
Comment (daneastmond) [reply src/Message/PurchaseRequest.php]: Done

**Pull request merged**